src/stravalib/client.py

```diff
@@ -189,17 +196,55 @@ def exchange_code_for_token(
             The developer client secret
         code : str
             The temporary authorization code
+        return_athlete : bool (default = False)
+            Whether to return a SummaryAthlete object (or not)
+            This parameter is currently undocumented and could change
+            at any time.
 
         Returns
         -------
-        dict
-            Dictionary containing the access_token, refresh_token and
+        AccessInfo
+            TypedDictionary containing the access_token, refresh_token and
             expires_at (number of seconds since Epoch when the provided access
             token will expire)
+        tuple
+            Contains:
+             - the AccessInfo typed dict containing token values
+             - a `SummaryAthlete` object representing the authenticated user.
+
+        Notes
+        -----
+        Strava by default returns `SummaryAthlete` information during
+        this exchange. However this return is currently undocumented
+        and could change at any time.
         """
-        return self.protocol.exchange_code_for_token(
-            client_id=client_id, client_secret=client_secret, code=code
+        raw = self.protocol.exchange_code_for_token(
```

Comment:
Can you try to unpack the return here? So
```python
access_info, athlete_data = self.protocol.exchange_code_for_token()
```
You can then return the tuple if `athlete_data and return_athlete == True`. This prevents the check on `len(raw)` and handling the `ValidationError` in case of a `None` response.